testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/security/AuthenticationPolicyContextTestCase.java

```diff
@@ -0,0 +1,333 @@
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/web.xml"), "web.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-deployment-structure.xml"), "jboss-deployment-structure.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-webservices.xml"), "jboss-webservices.xml");
+        return war;
+    }
+
+    @Deployment(name = PICKETLINK_STS_WS, managed = false, testable = false)
+    @TargetsContainer(CONTAINER)
+    public static Archive<?> createPicketlinkStsWsDeployment() {
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        WebArchive war = ShrinkWrap.create(WebArchive.class, PICKETLINK_STS_WS + ".war");
+        war.addClasses(EchoServiceRemote.class);
+        war.addClasses(EchoService.class);
+        war.addAsResource(tccl.getResource( RESOURCE_DIR + "sp-users.properties"), "sp-users.properties");
+        war.addAsResource(tccl.getResource( RESOURCE_DIR + "sp-roles.properties"), "sp-roles.properties");
+        war.addAsResource(tccl.getResource( RESOURCE_DIR + "sts-config.properties"), "sts-config.properties");
+        war.addAsWebInfResource(createFilteredAsset("resources/WEB-INF/wsdl/PicketLinkSTS.wsdl"), "wsdl/PicketLinkSTS.wsdl");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/beans.xml"), "beans.xml");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/jboss-web-ws.xml"), "jboss-web-ws.xml");
+        war.addAsWebInfResource(tccl.getResource( RESOURCE_DIR + "WEB-INF/jboss-wsse-server.xml"), "jboss-wsse-server.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-deployment-structure.xml"), "jboss-deployment-structure.xml");
+        war.addAsManifestResource(tccl.getResource(RESOURCE_DIR + "META-INF/jboss-webservices.xml"), "jboss-webservices.xml");
+        war.addAsResource(AuthenticationPolicyContextTestCase.class.getPackage(), "dummmy-ws-handler.xml", "org/jboss/as/test/manualmode/security/resources/dummmy-ws-handler.xml");
+        return war;
+    }
+
+    private static StringAsset createFilteredAsset(String resourceName) {
+        return new StringAsset(replaceNodeAddress(resourceName));
+    }
+
+    private static String replaceNodeAddress(String resourceName) {
+        String content = null;
+        try {
+            content = IOUtils.toString(AuthenticationPolicyContextTestCase.class.getResourceAsStream(resourceName), "UTF-8");
+        } catch (IOException e) {
+            throw new RuntimeException("Exception during replacing node address in resource", e);
+        }
+        return content.replaceAll("@node0@", NetworkUtils.formatPossibleIpv6Address(System.getProperty("node0", "127.0.0.1")));
+    }
+
+    @Before
+    public void before() {
+        container.start(CONTAINER);
```

Comment:
This test really seems to be doing a lot of things the hard way. I don't see why this can't use a normal ServerSetupTask+managed deployments to set up the security domains and do the deployments?